app/controllers/admin/users_controller.rb

```diff
@@ -3,9 +3,18 @@ class Admin::UsersController < ApplicationController
   before_action :check_admin
 
   def index
-    @users = User.all
+    if params[:waiting_approval]
+      @users = User.where.not(approved: true)
+    else
+      @users = User.all
+    end
   end
   
+  def approve
+    User.find(params[:id]).update(approved: true)
```

Comment:
def approve
    User.find(params[:id]).update(approved: true)
    redirect_to admin_users_path(waiting_approval: true), notice: 'User Approved'
  rescue ActiveRecord::RecordNotFound
    redirect_to admin_users_path, notice: 'Usen not found'
  end